Code/GraphMol/SmilesParse/SmilesWrite.cpp

```diff
@@ -466,35 +460,30 @@ std::string MolToSmiles(const ROMol &mol, bool doIsomericSmiles, bool doKekule,
       std::cout << "----------------------------" << std::endl;
 #endif
 
-    // adding randomness without setting the rootedAtAtom
-    if (doRandom) {
-      if (rootedAtAtom == -1) {
-        // need to find an atom id between 0 and mol.getNumAtoms() exclusively
-        rootedAtAtom = std::rand() % mol.getNumAtoms();
-      }
+    if (doRandom && rootedAtAtom == -1) {
+      // need to find a random atom id between 0 and mol.getNumAtoms()
+      // exclusively
+      rootedAtAtom = std::rand() % mol.getNumAtoms();
     }
 
     std::string res;
     unsigned int nAtoms = tmol->getNumAtoms();
-    UINT_VECT ranks(nAtoms);
+    std::vector<unsigned int> ranks(nAtoms);
```

Comment:
nice. I personally find this much more readable.

Reply:
yeah, but back in the day when we were chiseling the code into stone tablets it was important to save those characters!
;-)